src/test/kotlin/org/rust/ide/inspections/lints/RsPathStatementsInspectionTest.kt

```diff
@@ -0,0 +1,69 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.lints
+
+import org.rust.ProjectDescriptor
+import org.rust.WithStdlibRustProjectDescriptor
+import org.rust.ide.inspections.RsInspectionsTestBase
+
+class RsPathStatementsInspectionTest : RsInspectionsTestBase(RsPathStatementsInspection::class) {
+    fun `test path statement with no effect`() = checkByText("""
+        fn main() {
+            let n = 0;
+            <weak_warning descr="Path statement with no effect">n</weak_warning>;
+        }
+    """)
+
+    fun `test path statement with no effect in block`() = checkByText("""
+        fn main() {
+            let n = 0;
+            let _m = {
+                <weak_warning descr="Path statement with no effect">n</weak_warning>;
+                n
+            };
+        }
+    """, ignoreExtraHighlighting = false)
+
+    fun `test fix by removing statement`() = checkFixByText("Remove `n;`", """
+        fn main() {
+            let n = 0;
+            /*caret*/<weak_warning descr="Path statement with no effect">n</weak_warning>;
+        }
+    """, """
+        fn main() {
+            let n = 0;
+        /*caret*/}
+    """)
+
+    fun `test path statement drop`() = checkByText("""
+        struct DropMe;
+        impl Drop for DropMe { fn drop(&mut self) {} }
+        fn main() {
+            let a = DropMe;
+            <weak_warning descr="Path statement drops value">a</weak_warning>;
+
+            let b = (DropMe,);
+            <weak_warning descr="Path statement drops value">b</weak_warning>;
+        }
+    """)
+
+    @ProjectDescriptor(WithStdlibRustProjectDescriptor::class)
```

Comment:
Looks like we could use a stub `Drop` definition like this to avoid heavy `WithStdlibRustProjectDescriptor`
```Rust
#[lang = "drop"]
pub trait Drop {
    fn drop(&mut self);
}
```